src/mmds/src/data_store.rs

```diff
@@ -386,5 +402,31 @@ mod tests {
         }"#;
         let data_store: Value = serde_json::from_str(data).unwrap();
         assert!(mmds.patch_data(data_store).is_ok());
+
+        let filling = (0..51151).map(|_| "X").collect::<String>();
+        let data = "{\"new_key\": \"".to_string() + &filling + "\"}";
+
+        let data_store: Value = serde_json::from_str(&data).unwrap();
+        assert!(mmds.patch_data(data_store).is_ok());
+
+        let data = "{\"new_key2\" : \"smth\"}";
+        let data_store: Value = serde_json::from_str(&data).unwrap();
+        assert_eq!(
+            mmds.patch_data(data_store).unwrap_err(),
+            Error::DataStoreLimitExceeded
+        );
+        assert!(!mmds.get_data_str().contains("smth"));
+
+        let data = "{\"new_key\" : \"smth\"}";
+        let data_store: Value = serde_json::from_str(&data).unwrap();
+        assert!(mmds.patch_data(data_store).is_ok());
+        assert!(mmds.get_data_str().contains("smth"));
+        assert_eq!(mmds.get_data_str().len(), 53);
+
+        let data = "{\"new_key2\" : \"smth2\"}";
+        let data_store: Value = serde_json::from_str(&data).unwrap();
+        assert!(mmds.patch_data(data_store).is_ok());
+        assert!(mmds.get_data_str().contains("smth2"));
+        assert_eq!(mmds.get_data_str().len(), 72);
```

Comment:
I think we can get rid of this, as it doesn't test a different case than the part above and I find it redundant.

Reply:
it is not redundant, it actually tests that you can add new data once there is enough space in data_store, in this case particularly after we delete some data with a patch we are able to add new data.

Reply:
I see your point. This sounds good.